Semver/Ranges/Comparers/Npm/Comparator.cs

```diff
@@ -0,0 +1,254 @@
+using System;
+using System.Linq;
+using System.Text;
+using Semver.Comparers;
+
+namespace Semver.Ranges.Comparers.Npm
+{
+    internal class NpmComparator : IEquatable<NpmComparator>
+    {
+        public readonly ComparatorOp Operator;
+        public readonly SemVersion Version;
+        public readonly bool AnyVersion;
+
+        private readonly NpmParseOptions options;
+
+        private NpmComparator[] comparators; // The other comparators in the same AND range
+        private string cachedStringValue;
+
+        public NpmComparator(ComparatorOp @operator, SemVersion version, NpmParseOptions options)
+        {
+            if (@operator == ComparatorOp.ReasonablyClose || @operator == ComparatorOp.CompatibleWith)
+                throw new ArgumentException("Invalid operator (ReasonablyClose and CompatibleWith are invalid uses in this context)");
+            
+            Operator = @operator;
+            Version = version;
+            this.options = options;
+        }
+
+        /// <summary>
+        /// Any version will match when using this constructor
+        /// </summary>
+        public NpmComparator(NpmParseOptions options)
+        {
+            AnyVersion = true;
+            this.options = options;
+        }
+
+        internal void SetRangeComparators(NpmComparator[] comparators)
```

Comment:
You don't actually need the other comparators right? You just need to know whether they should cause prerelease to be included. I would just set that here and avoid recalculating it each time the comparison is checked.

Reply:
I'm not sure i understand what you mean. The prerelease check only applies if the input version parameter's main version matches another component's main version.